### DISCUSSION TITLE
React-Markdown upgrade error

### DISCUSSION THREAD

**Discussion by chulinguy** [remarkjs/remark#1299] [Q&A]
Hi, 

For my project, I am currently on:
- `react-markdown` 6.0.3
- `remark-gfm` 1.0.0
- `node` 18.19.0

I would like to upgrade both `react-markdown` and `remark-gfm`, but when I do, my markdown components show this strange error.  Can someone give some pointers please?

<img width="1023" alt="Screenshot 2024-03-15 at 4 28 27 PM" src="https://github.com/remarkjs/remark/assets/25993366/a8d54b3b-3006-484b-96bd-603b9ec09310">

**Comment by ChristianMurphy** [comment]
Welcome @chulinguy! 👋 
Sorry you ran into a spot of trouble.

When you update, make sure to update `react-markdown`, `remark-gfm`, and any other `remark` plugins you use to their latest version at the same time.

The error you were seeing indicates you were trying to mix major versions several years different from each other.

**Reply by chulinguy** [reply]
I tried to go to `react-markdown` 9.0.1 and `remark-gfm` 4.0.0 together, those are the most recent ones now.  
I don't have other direct `remark` dependencies in the codebase.  

Do you recommend a different pairing?

**Reply by ChristianMurphy** [reply]
> those are the most recent ones now.
> Do you recommend a different pairing?

No, use the most recent for each.

---

Have you followed the migration guide for each release?
https://github.com/remarkjs/react-markdown/blob/main/changelog.md#changelog
https://github.com/remarkjs/remark-gfm/releases

If so, could you show how you are using the libraries together in a code snippet or codesandbox?
Including information about what build tool(s) you are using.
